lib/lltz_ir/dsl.ml

```diff
@@ -0,0 +1,183 @@
+(* dsl.ml *)
+open Grace
+module LE = Expr
+module LT = Type
+module LP = Primitive
+
+module T = struct
+  type var = Var of string
+  type mut_var = Mut_var of string
+
+  let dummy : Range.t = Range.initial (`String { content = ""; name = Some "" })
+
+  (* Constants *)
+  let unit = LE.{ desc = LE.Const(Unit); range = dummy }
```

Comment:
May I suggest you define a function `let mk_dummy desc = LE.{desc; range=dummy}`?

Reply:
Agree. Additionally suggest `?range` optional arguments in functions to allow us to explicitly pass locations